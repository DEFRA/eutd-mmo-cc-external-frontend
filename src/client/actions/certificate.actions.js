import { isError, showFullPageError } from './index';

export const GET_SUMMARY_DOCUMENT = 'GET_SUMMARY_DOCUMENT';
export const SHOW_SUMMARY_DOCUMENT_ERRORS = 'SHOW_SUMMARY_DOCUMENT_ERRORS';
export const GET_SUMMARY_UNAUTHORISED = 'GET_SUMMARY_DOCUMENT_UNAUTHORISED';

export function showSummaryCertificateErrors(data) {
  return { type: SHOW_SUMMARY_DOCUMENT_ERRORS, payload: { validationErrors: data } };
}

export const getSummaryCertificate = (journey, documentNumber = undefined) => async (dispatch, getState, { orchestrationApi }) => {
  const config = (documentNumber)
    ? { headers: { documentnumber: documentNumber } }
    : {};

  await orchestrationApi.get(`/certificate/${journey}`, config)
    .then(
      res => {
        dispatch({ type: GET_SUMMARY_DOCUMENT, payload: res.data });
      },
      error => {
        if (isError(error)) {
          dispatch(showFullPageError());
        } else {
          dispatch(unauthorised());
        }
      });

    function unauthorised() { return { type: GET_SUMMARY_UNAUTHORISED }; }
};