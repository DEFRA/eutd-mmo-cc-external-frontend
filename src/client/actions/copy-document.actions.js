import { showFullPageError, apiCallFailed, isError, CLEAR_ERRORS } from './index';
export const CONFIRM_COPY_DOCUMENT            = 'confirm_copy_document';
export const UNAUTHORISED_COPY_DOCUMENT       = 'unauthorised_copy_document';
export const CLEAR_COPY_DOCUMENT              = 'clear_copy_document';
export const REMOVE_CONFIRM_COPY_CERTIFICATE  = 'remove_confirm_copy_certificate';
export const ADD_ERROR_COPY_CERTIFICATE       = 'add_error_copy_certificate';
export const REMOVE_ERROR_COPY_CERTIFICATE    = 'remove_error_copy_certificate';

export const submitCopyCertificate = (payload) => async (dispatch, _getState, {orchestrationApi}) => {
  const {documentNumber, copyDocumentAcknowledged, voidOriginal, excludeLandings, journey} = payload;
  const config = {headers: {documentnumber: documentNumber}};

  try {
    let ipAddress;
    if (voidOriginal) {
      const ip = await orchestrationApi.get('/client-ip');
      ipAddress = ip.data;
    }

    return await orchestrationApi.post('/confirm-copy-certificate', { copyDocumentAcknowledged, voidOriginal, excludeLandings, journey, ipAddress }, config)
    .then(res => {
      dispatch({ type: CONFIRM_COPY_DOCUMENT, payload: {
        documentNumber: res.data.documentNumber,
        copyDocumentAcknowledged: res.data.copyDocumentAcknowledged,
        voidDocumentConfirm: res.data.voidOriginal
      }});
      return res.data.newDocumentNumber;
    });
  } catch (e) {
    if (isError(e)) dispatch(showFullPageError());
    if (e.response && e.response.status === 403) {
      dispatch({ type: UNAUTHORISED_COPY_DOCUMENT }); return;
    } else {
      dispatch(apiCallFailed(e.response));
    }
  }
};

export const checkCopyCertificate = (documentNumber) => async (dispatch, _getState, {orchestrationApi}) => {
  const config = {
    headers: {
      documentnumber: documentNumber
    }
  };

  try {
    const res = await orchestrationApi.get('/check-copy-certificate', config);
    const { canCopy } = res.data;
    if (!canCopy) {
      dispatch(unauthorised());
    }
  } catch(e) {
    if (isError(e)) dispatch(showFullPageError());
    if (e.response && e.response.status === 403) {
      dispatch(unauthorised());
      throw new Error(`Unauthorised access ${e.response.status}`);
    } else {
      dispatch(apiCallFailed(e.response));
      throw new Error(`An error has occurred ${e.response.status} (${e.response.statusText})`);
    }
  }

  function unauthorised() { return { type: UNAUTHORISED_COPY_DOCUMENT }; }
};

export const addCopyDocument = (data) => (dispatch) => dispatch({ type: CONFIRM_COPY_DOCUMENT, payload: data });
export const clearErrors = () => (dispatch) =>  dispatch({ type: CLEAR_ERRORS });
export const clearCopyDocument = () => (dispatch) => dispatch({ type: CLEAR_COPY_DOCUMENT });
export const unauthorisedCopyDocument = () => (dispatch) => dispatch({ type: UNAUTHORISED_COPY_DOCUMENT });