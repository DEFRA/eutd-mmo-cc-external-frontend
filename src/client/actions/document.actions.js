// actions
import {isError, showFullPageError} from './index';

export const GOT_ALL_DOCUMENTS          = 'GOT_ALL_DOCUMENTS';
export const GET_ALL_DOCUMENTS_FAILED   = 'GET_ALL_DOCUMENTS_FAILED';
export const GET_COMPLETED_DOCUMENT = 'GET_COMPLETED_DOCUMENT';
export const GET_PENDING_DOCUMENT = 'GET_PENDING_DOCUMENT';


// action creators
export const getAllDocuments = (type, month, year) => async (dispatch, getState, { orchestrationApi }) => {
  try {
    const res = await orchestrationApi.get(`/documents/${year}/${month}?type=${type}`);
    return dispatch({ type: GOT_ALL_DOCUMENTS, payload: res.data });

  } catch(e) {
    if( isError(e) ) dispatch(showFullPageError());
    //throw new Error(e.response);
  }
};

export const getCompletedDocument = (documentNumber) => async (dispatch, getState, { orchestrationApi }) => {
  try {
    const res = await orchestrationApi.get(`/document/${documentNumber}`);
    if (res.data && res.data.documentStatus === 'COMPLETE') {
      return dispatch({ type: GET_COMPLETED_DOCUMENT, payload: res.data });
    }
    return dispatch({ type: GET_COMPLETED_DOCUMENT, payload: null });
  }
  catch (e) {
    return dispatch({ type: GET_COMPLETED_DOCUMENT, payload: null });
  }
};

export const getPendingDocument = (documentNumber) => async (dispatch, getState, { orchestrationApi }) => {
  try {
    const res = await orchestrationApi.get(`/document/${documentNumber}`);
    if (res.data && res.data.documentStatus === 'PENDING') {
      return dispatch({ type: GET_PENDING_DOCUMENT, payload: res.data });
    }
    return dispatch({ type: GET_PENDING_DOCUMENT, payload: null });
  }
  catch (e) {
    return dispatch({ type: GET_PENDING_DOCUMENT, payload: null });
  }
};