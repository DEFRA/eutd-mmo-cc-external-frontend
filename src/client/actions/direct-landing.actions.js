import { beginApiCall, apiCallFailed, isForbidden, showFullPageError, CLEAR_ERRORS, isSilverLineError } from './index';
import { exportPayloadActionTypes } from '../actions';


export const GET_DIRECT_LANDING_PRODUCTS = 'get_direct_landing_products';
export const CLEAR_DIRECT_LANDING = 'clear_direct_landing';
export const DIRECT_LANDING_PRODUCTS_FAILED = 'direct_landing_products_failed';
export const POST_DIRECT_LANDING_PRODUCTS_UNAUTHORISED = 'post_direct_landing_products_unauthorised';
export const GET_DIRECT_LANDING_PRODUCTS_UNAUTHORISED = 'get_direct_landing_products_unauthorised';
export const POST_DIRECT_LANDING_PRODUCTS_WAF_ERROR = 'post_direct_landing_prodcts_waf_error';

export const getDirectLandings =
  (documentNumber) =>
  async (dispatch, _getState, { orchestrationApi }) => {
    const config = { headers: { documentnumber: documentNumber } };

    try {
      dispatch(beginApiCall());

      const res = await orchestrationApi.get(
        '/export-certificates/export-payload/direct-landings',
        config
      );

      await dispatch({ type: GET_DIRECT_LANDING_PRODUCTS, payload: res.data });
    } catch (e) {
      if (isForbidden(e)) {
        dispatch({ type: GET_DIRECT_LANDING_PRODUCTS_UNAUTHORISED });
      } else {
        dispatch(apiCallFailed(e.response));
        throw new Error(
          `An error has occurred ${e.response.status} (${e.response.statusText})`
        );
      }
    }
  };

export const clearDirectLanding = () => ({ type: CLEAR_DIRECT_LANDING });

export const upsertDirectLanding = (landing, documentNumber = undefined) => (dispatch, getState, { orchestrationApi } ) => {
  const config = (documentNumber)
    ? {
        headers: {
          documentnumber: documentNumber
        }
    }
    : {};

  return orchestrationApi.post('/export-certificates/direct-landing/validate', landing, config)
    .then(
      res => {
        dispatch(landingUpserted(res.data));
        dispatch(clearErrors());
      },
      error => {
        if (error.response.status === 400) {
          dispatch(landingUpserted(error.response.data));
          throw new Error();
        } else if (isSilverLineError(error)) {
          dispatch({ type: POST_DIRECT_LANDING_PRODUCTS_WAF_ERROR, supportID: error.response.data.match(/\d+/)[0] });
        } else if (isForbidden(error)) {
          dispatch({ type: POST_DIRECT_LANDING_PRODUCTS_UNAUTHORISED });
        } else {
          dispatch(showFullPageError());
        }
      });

  function landingUpserted(exportPayload) { return { type: exportPayloadActionTypes.LANDING_UPSERTED, exportPayload, errors: exportPayload.errors }; }
  function clearErrors() { return { type: CLEAR_ERRORS }; }
};
