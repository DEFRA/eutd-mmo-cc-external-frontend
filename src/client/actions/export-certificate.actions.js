import { showSummaryCertificateErrors } from './certificate.actions';

export const exportCertificateActionTypes = {
  EXPORT_CERT_CREATE_REQUEST: 'export-certificate/create',
  EXPORT_CERT_CREATE_REQUEST_SUCCESS: 'export-certificate/create/success',
  EXPORT_CERT_CREATE_REQUEST_FAILED: 'export-certificate/create/failed'
};

export const createExportCertificate = (currentUri, journey, noOfVessels, documentNumber) => async (dispatch, getState, { orchestrationApi }) => {
  const config = (documentNumber)
    ? {headers: {documentNumber: documentNumber}}
    : {};

  const { data } = await orchestrationApi.get('/client-ip');

  return await orchestrationApi.post('/export-certificates/create', { journey, data },config)
    .then(
      res => {
        let certDetails = res.data;
        if (certDetails.status && certDetails.status === 'invalid catch certificate') {
          return 'ERROR';
        }
        window.dataLayer = window.dataLayer || [];
        window.dataLayer.push({
          'event': 'completedCatchCertificate',
          'numberOfVessels': noOfVessels
        });
        return dispatch(exportCertificateCreated(certDetails));
      },
      error => {
        dispatch(exportCertificateCreateFailed(error.toString()));
        if(error.response && error.response.data) {
          return dispatch(showSummaryCertificateErrors(error.response.data));
        }
      })
    .catch((err) => {
      dispatch(exportCertificateCreateFailed(err));
      throw(err);
    });

  function exportCertificateCreated(certDetails) { return { type: exportCertificateActionTypes.EXPORT_CERT_CREATE_REQUEST_SUCCESS,  payload: { data: certDetails } }; }
  function exportCertificateCreateFailed(errorRes) { return { type: exportCertificateActionTypes.EXPORT_CERT_CREATE_REQUEST_FAILED, errorRes }; }
};

export const getExportCertificateFromParams = (params) => (dispatch) => {
  return dispatch({type: exportCertificateActionTypes.EXPORT_CERT_CREATE_REQUEST_SUCCESS, payload: {data: params }});
};