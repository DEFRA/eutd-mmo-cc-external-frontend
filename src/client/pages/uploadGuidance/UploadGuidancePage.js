import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { getAllUserAttributes } from '../../actions/userAttributes.actions';
import { UploadGuidanceEnglish } from './UploadGuidanceEnglish';
import { UploadGuidanceWelsh } from './UploadGuidanceWelsh';

export const UploadGuidancePage = (props) => {
  let userSelectedLanguage;
  const userAttributeLanguage = props.userAttributes.find((val) => val.name === 'language');

  if (userAttributeLanguage) {
    userSelectedLanguage = userAttributeLanguage.value;
  }
  return (
    userSelectedLanguage === 'cy_UK'
      ? <UploadGuidanceWelsh {...props} />
      : <UploadGuidanceEnglish {...props} />
  );
};

const mapStateToProps = state => {
  return {
    userAttributes: state.userAttributes
  };
};

const loadData = store => {
  return store.dispatch(getAllUserAttributes());
};

export default {
  loadData,
  component: withRouter(connect(mapStateToProps,
    {
      getAllUserAttributes,
    })(UploadGuidancePage)),
};
