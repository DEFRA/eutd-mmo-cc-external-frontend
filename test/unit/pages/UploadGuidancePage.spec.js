import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import thunk from 'redux-thunk';
import * as UploadGuidancePage from '../../../src/client/pages/uploadGuidance/UploadGuidancePage';
import UploadGuidancePageWrapper from '../../../src/client/pages/uploadGuidance/UploadGuidancePage';
import { getAllUserAttributes } from '../../../src/client/actions/userAttributes.actions';

jest.mock('../../../src/client/actions/userAttributes.actions');

describe('UploadGuidancePage', () => {
  let wrapper;
  const mockStore = configureStore([thunk]);

  const getWrapper = (selectedLanguage) => {
    const store = mockStore({
      userAttributes: [{ name: 'language', value: selectedLanguage }],
    });

    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <UploadGuidancePage.default.component />
        </MemoryRouter>
      </Provider>
    );

    wrapper = container;

    return wrapper;
  };

  it('should render an UploadGuidancePage component', () => {
    const wrapper = getWrapper();
    expect(wrapper).toBeDefined();
  });

  it('should render the page in English', () => {
    const wrapper = getWrapper('en_UK');
    expect(wrapper).toBeDefined();
  });

  it('should render the page in Welsh', () => {
    const wrapper = getWrapper('cy_UK');
    expect(wrapper).toBeDefined();
  });

  it('should render the page in English if no language has been selected', () => {
    const store = mockStore({
      userAttributes: [],
    });

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <UploadGuidancePage.default.component />
        </MemoryRouter>
      </Provider>
    );

    expect(wrapper).toBeDefined();
    expect(
      wrapper.find('PageTitle[title=\'Upload Guidance - GOV.UK\']').exists()
    ).toBeTruthy();
  });

  it('should take a snapshot of the whole page', () => {
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call getAllUserAttributes action to load the component', () => {
    const store = { dispatch: jest.fn() };

    getAllUserAttributes.mockReturnValue({ type: 'GET_ALL_USER_ATTRIBUTES' });

    UploadGuidancePageWrapper.loadData(store);

    expect(getAllUserAttributes).toHaveBeenCalled();
    expect(getAllUserAttributes).toHaveBeenCalledTimes(1);
  });
});
