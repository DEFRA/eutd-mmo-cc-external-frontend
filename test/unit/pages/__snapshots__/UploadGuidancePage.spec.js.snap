// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadGuidancePage should take a snapshot of the whole page 1`] = `
<div>
  <div
    class="css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <a
            class="e1vcssft0 css-zk98nb-Anchor-StyledHoc e4u1u2g0"
            href="#"
          >
            Back
          </a>
          <h1
            class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
          >
            Upload guidance
          </h1>
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
            data-testid="data-general-label"
          >
            General
          </h2>
          <table
            class="css-1uqv7s3-StyledTable e4u1u2g0"
            data-testid="data-general-table"
          >
            <caption
              class="css-gzlitc-Caption e172z3n50"
            />
            <thead
              class="css-1iuscil-TableHeader e1tbxulj0"
            >
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <th
                  class="css-12e3svi-CellHeader e1sike1z0"
                >
                  Subject
                </th>
                <th
                  class="css-12e3svi-CellHeader e1sike1z0"
                >
                  Guidance
                </th>
              </tr>
            </thead>
            <tbody
              class="css-x1p7io-TableBody eo9ngsf0"
            >
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                  style="vertical-align: top; text-align: left; border-bottom: 1px solid #b1b4b6; padding: 10px 20px 10px 0px;"
                >
                  <strong>
                    Upload process
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <ul
                    class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
                  >
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      The steps of the upload process are as follows:
                      <ol
                        class="ol e1vcssft0 css-1oae491-StyledList-StyledHoc e4u1u2g0"
                      >
                        <li
                          class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                        >
                           
                          <strong>
                            Upload products and landings
                          </strong>
                           - Upload the CSV file containing the product IDs and landings.
                        </li>
                        <li
                          class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                        >
                           
                          <strong>
                            Fix errors (if required)
                          </strong>
                           - If there are multiple errors, you can clear the upload, amend the file, and reupload.
                        </li>
                        <li
                          class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                        >
                           
                          <strong>
                            Review/correct
                          </strong>
                            - The steps will take you back to the Add landings page where you can add any missing landings manually or correct any mistakes you find.
                        </li>
                      </ol>
                    </li>
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      Multiple CSV files can be uploaded by cycling through the process again.
                    </li>
                  </ul>
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                  style="vertical-align: top; text-align: left; border-bottom: 1px solid #b1b4b6; padding: 10px 20px 10px 0px;"
                >
                  <strong>
                    Limitations
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <ul
                    class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
                  >
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      The maximum number of rows that can be uploaded per catch certificate is 100.
                    </li>
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      You can only upload products that have been saved in your product favourites, accessible from the main navigation.
                    </li>
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      If your export contains multiple products from the same landing, the landing will need to be repeated for each product.
                    </li>
                  </ul>
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                  style="vertical-align: top; text-align: left; border-bottom: 1px solid #b1b4b6; padding: 10px 20px 10px 0px;"
                >
                  <strong>
                    Product IDs
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <ul
                    class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
                  >
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      Product IDs are created when a  
                      <a
                        aria-label="Opens link for information on product favourites"
                        href="/manage-favourites"
                      >
                        <span
                          class="govuk-visually-hidden"
                        >
                          (opens in same tab)
                        </span>
                        product favourites
                      </a>
                        is saved.
                    </li>
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      They are unique to each user and are used to speed up the process of uploading products.
                    </li>
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      Product IDs are only used internally by the FES service and have no relevance to the final catch certificate.
                    </li>
                  </ul>
                </td>
              </tr>
            </tbody>
          </table>
          <br />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
            data-testid="data-csvfile-label"
          >
            CSV file
          </h2>
          <table
            class="css-1uqv7s3-StyledTable e4u1u2g0"
            data-testid="data-csvfile-table"
          >
            <caption
              class="css-gzlitc-Caption e172z3n50"
            />
            <thead
              class="css-1iuscil-TableHeader e1tbxulj0"
            >
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <th
                  class="css-12e3svi-CellHeader e1sike1z0"
                >
                  Subject
                </th>
                <th
                  class="css-12e3svi-CellHeader e1sike1z0"
                >
                  Guidance
                </th>
              </tr>
            </thead>
            <tbody
              class="css-x1p7io-TableBody eo9ngsf0"
            >
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                  style="vertical-align: top; text-align: left; border-bottom: 1px solid #b1b4b6; padding: 10px 20px 10px 0px;"
                >
                  <strong>
                    File type
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <ul
                    class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
                  >
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      The upload file must be a CSV file (CSV stands for Comma Separated Values).
                    </li>
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      CSV files can be generated by exporting from spreadsheets or other software with an export feature.
                    </li>
                  </ul>
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                  style="vertical-align: top; text-align: left; border-bottom: 1px solid #b1b4b6; padding: 10px 20px 10px 0px;"
                >
                  <strong>
                    Data structure
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <ul
                    class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
                  >
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      Do not include a header row.
                    </li>
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      Each row should have the following structure:
                      <div
                        class="e1vcssft0 css-faqcc9-StyledInsetText-StyledHoc e4u1u2g0"
                        style="margin-top: 30px;"
                      >
                        <strong>
                          Product ID,Date landed,Catch area,Vessel PLN,Export weight
                        </strong>
                         
                        <br />
                        For example: PRD123,01/01/2021,FAO27,PH1100,50.95
                      </div>
                    </li>
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      Only use commas to separate fields (known as 'delimiters').
                    </li>
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      You are advised to group the rows together in a way that makes it easy for you to check the upload. For example, you could group landings by product or by vessel.
                    </li>
                  </ul>
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                  style="vertical-align: top; text-align: left; border-bottom: 1px solid #b1b4b6; padding: 10px 20px 10px 0px;"
                >
                  <strong>
                    Validation
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <ul
                    class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
                  >
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      Uploaded products and landings will be subjected to the same validation as those entered manually.
                    </li>
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      If some rows fail to upload and you want to add them later manually, you should make a note of them before continuing.
                    </li>
                  </ul>
                </td>
              </tr>
            </tbody>
          </table>
          <br />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
            data-testid="data-requirements-label"
          >
            Data requirements
          </h2>
          <table
            class="css-1uqv7s3-StyledTable e4u1u2g0"
            data-testid="data-requirements-table"
          >
            <caption
              class="css-gzlitc-Caption e172z3n50"
            />
            <thead
              class="css-1iuscil-TableHeader e1tbxulj0"
            >
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <th
                  class="css-12e3svi-CellHeader e1sike1z0"
                >
                  Field
                </th>
                <th
                  class="css-12e3svi-CellHeader e1sike1z0"
                >
                  Guidance
                </th>
                <th
                  class="css-12e3svi-CellHeader e1sike1z0"
                >
                  Example
                </th>
              </tr>
            </thead>
            <tbody
              class="css-x1p7io-TableBody eo9ngsf0"
            >
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                  style="vertical-align: top; text-align: left; border-bottom: 1px solid #b1b4b6; padding: 10px 20px 10px 0px;"
                >
                  <strong>
                    Product ID
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <ul
                    class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
                  >
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      The product ID must refer to a product saved in your product favourites.
                    </li>
                  </ul>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  PRD123
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                  style="vertical-align: top; text-align: left; border-bottom: 1px solid #b1b4b6; padding: 10px 20px 10px 0px;"
                >
                  <strong>
                    Date landed
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <ul
                    class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
                  >
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      Landing dates must be real dates in the format 'dd/mm/yyyy'.
                    </li>
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      Dates cannot be more than 7 days in the future.
                    </li>
                  </ul>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  01/01/2021
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                  style="vertical-align: top; text-align: left; border-bottom: 1px solid #b1b4b6; padding: 10px 20px 10px 0px;"
                >
                  <strong>
                    Catch area
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <ul
                    class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
                  >
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      The catch area must be a FAO major fishing area.
                    </li>
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      A list of the major fishing areas is available from: www.fao.org
                    </li>
                  </ul>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  FAO27
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                  style="vertical-align: top; text-align: left; border-bottom: 1px solid #b1b4b6; padding: 10px 20px 10px 0px;"
                >
                  <strong>
                    Vessel PLN
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <ul
                    class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
                  >
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      The PLN must be a valid Vessel PLN.
                    </li>
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      Vessels must be licensed on the date the catch was landed.
                    </li>
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      Vessel PLNs are available from: www.fishhub.cefas.co.uk/vessel-register/
                    </li>
                  </ul>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  PH1100
                </td>
              </tr>
              <tr
                class="css-7u9vzt-RowInner e1bh3ka20"
              >
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                  style="vertical-align: top; text-align: left; border-bottom: 1px solid #b1b4b6; padding: 10px 20px 10px 0px;"
                >
                  <strong>
                    Export weight
                  </strong>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  <ul
                    class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
                  >
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      The export weight must be in kilograms (kg).
                    </li>
                    <li
                      class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
                    >
                      Weights must be greater than zero with up to two decimal places.
                    </li>
                  </ul>
                </td>
                <td
                  class="css-14ci18q-CellInner edqrv1d0"
                >
                  50.95
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <hr
            class="help-link"
          />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Need help?
          </h2>
          <a
            href="undefined"
            rel="noopener noreferrer"
            target="_blank"
          >
            Get help exporting fish from the UK (gov.uk)
             
            <span
              class="govuk-visually-hidden"
            >
              (opens in new tab)
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
